src/ripple/app/rdb/impl/RelationalDBInterface_shards.cpp

```diff
@@ -32,6 +32,121 @@
 
 namespace ripple {
 
+DatabasePair
+makeMetaDBs(
+    Config const& config,
+    DatabaseCon::Setup const& setup,
+    DatabaseCon::CheckpointerSetup const& checkpointerSetup)
+{
+    // ledger meta database
+    auto lgr{std::make_unique<DatabaseCon>(
+        setup,
+        LgrMetaDBName,
+        LgrMetaDBPragma,
+        LgrMetaDBInit,
+        checkpointerSetup)};
+
+    if (config.useTxTables())
+    {
+        // transaction meta database
+        auto tx{std::make_unique<DatabaseCon>(
+            setup,
+            TxMetaDBName,
+            TxMetaDBPragma,
+            TxMetaDBInit,
+            checkpointerSetup)};
+
+        return {std::move(lgr), std::move(tx)};
+    }
+
+    return {std::move(lgr), {}};
+}
+
+bool
+saveLedgerMeta(
+    std::shared_ptr<Ledger const> const& ledger,
+    Application& app,
+    soci::session& lgrSesh,
+    soci::session& txnSesh,
+    std::uint32_t const shardIndex)
```

Comment:
I am wondering if it would be better to use a `DatabasePair` ref here instead of the session params. That would allow the function to only created locked sessions when/if necessary.

Consider renaming `lgrSesh` to `lgrSession` or `lgrMetaSession`

Reply:
Renamed. Using a `DatabasePair` would create some challenges since it's holding `unique_ptr`s.